tests/api_resources/test_file_upload.py

```diff
@@ -0,0 +1,57 @@
+import tempfile
+
+import pytest
+
+import stripe
+
+
+TEST_RESOURCE_ID = "file_123"
+
+
+class TestFileUpload(object):
+    @pytest.fixture(scope="function")
+    def setup_upload_api_base(self):
+        stripe.upload_api_base = stripe.api_base
+        stripe.api_base = None
+        yield
+        stripe.api_base = stripe.upload_api_base
+        stripe.upload_api_base = "https://files.stripe.com"
```

Comment:
Considering we are replacing the entire HTTP client. Why do we even need to mess with API bases? Can we always keep them default?

Reply:
You're right, we can keep them default. Updated in https://github.com/stripe/stripe-python/pull/1179/commits/54caa2faa8b457a49d909f5eb4759a02354e4e4e

Reply:
Nevermind, just realized `setup_upload_api_base` is necessary because it makes it so that `stripe.api_base = None`, so if `stripe.api_base` is mistakenly used and API base is asserted to be `stripe.upload_api_base`, the test will fail.

Reply:
go/j/DEVSDK-1430